web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -73,6 +73,25 @@ export class DataStoreService {
       .set({ title: { en: newTitle } }, { merge: true })
       .then(() => projectId);
   }
+
+  // TODO: Define return types for methods in this class
+  updateProjectLayer(projectId: string, layerId: string, layer: Layer) {
+    return new Promise((resolve, reject) => {
```

Comment:
Immediately before doing these you should copy `Layer` in an `any` and remove the `id` field

Reply:
I could have used any here but it would have given me typescript error and the build would have failed , so I opted to use { id: string } which is safer as we are expecting id as one of fields and removing it :)

Reply:
Cool trick, didn't know you could do this. Thanks!